### ISSUE TITLE
Ability to cancel a timer

### ISSUE THREAD

**Issue by JSti** [jeroenbos22/homey.countdown#20]
First thanks for this great Homey app. 

I'm trying to create some flows with this app, and miss the ability to distinguish expired timers from (programmatically) cancelled timers with the "stop timer" action. Right now I can only postpone a flow which triggers after a timer stops without using other tools like variables. 

So both stopping manually and the timer expiring ends in the same stop trigger, and I'd like some flows to only run when a timer actually runs out (expires) and not when I stop a timer manually. For instance, when I stop the timer from a different flow before it expires I'd like to prevent the regular behaviour "turn off a light" from happening which would normally happen when the times expires.

I think the easiest approach would be to add an extra condition. Somebody can create a flow as usual which triggers on "timer stopped" and add an extra condition "timer expired" to make sure it isn't stopped manually. By leaving this condition out it behaves like your app does now, and inverting this condition makes sure the flow only runs when a timer is cancelled using the "stop timer" action.

Another approach but probably much more complicated to implement is to add extra triggers instead of a condition. To illustrate, it could look something like this:

Triggers:
- Timer started
- Timer cancelled
- Timer expired
- Timer stopped (which includes both cancelled and expired timers)

Actions:
- Stop an existing timer: cancels a timer and triggers "timer cancelled" and "timer stopped". Could be the action as is now.
- (Optional) end/expire an existing timer, or just support this using the "update timer" action by updating an existing timer to 0 seconds. Should trigger both "timer expired" and "timer stopped".

Would something like this fit into the roadmap of your app? Or is this something I should solve differently.

**Comment by GreyEarl** [comment]
Roadmap

Write MORE tests
Mock Homey npm module
Add adding or substracting a fixed or random number of seconds
Ability to pause a timer 

The last one should cover it, right?

**Comment by JSti** [comment]
Not sure, might be usable for the same purposes but really depends on the planned implementation.

**Comment by mvl64** [comment]
I have a similar need - difference between a flow stopping/cancelling a timer (e.g. a "no motion" timer that needs to be cancelled at the moment somone comes in a room) and the timer actually reaching zero. It would be really nice if you can implement this.

I noticed this distinction is already available in the countdown app by Ralph van Dooren - so using that one for the timers that need the option of cancelling. I do however like your apps UI (particularly the fact I can see the timer countdown and option to manually pause/modify).

**Comment by Shakesbeard** [comment]
Yes. I would really second that. I too need separate events for if a timer has ended by countdown or if it has been stopped by an action.

**Comment by jeroenbos22** [comment]
Try the new test-version of the app on https://homey.app/nl-nl/app/homey.countdown/Countdown-timers/test/

This should fix your issue!
